### ISSUE TITLE
usrv init errors: table not created, + usrv.php not found

### ISSUE THREAD

**Issue by joshp23** [joshp23/YOURLS-U-SRV#6]
I decided to removed the usrv table as I created it manually, perhaps that's why the type was InnoDB and your other plugins were MyISAM. 
After enabling the usrv plugin, plugin page debug info:
```
Connected to database u12345p12345_yourls on localhost 
SQL: SELECT option_name, option_value FROM yourls_options WHERE 1=1 (0.00056 s)
SQL: SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES
    		WHERE TABLE_NAME = 'usrv'
    		AND ENGINE = 'MyISAM' LIMIT 1 (0.00636 s)
```

**Note**: NO usrv table is created in database!

Then I clicked on U-SRV plugin link:
```
Fatal error: Uncaught PDOException: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'u12345p12345_yourls.usrv' doesn't exist in /home/u12345p12345/domains/mydomain.com/public_html/includes/vendor/aura/sql/src/ExtendedPdo.php:748 Stack trace: #0 /home/u12345p12345/domains/mydomain.com/public_html/includes/vendor/aura/sql/src/ExtendedPdo.php(748): PDOStatement->execute() #1 /home/u12345p12345/domains/mydomain.com/public_html/includes/vendor/aura/sql/src/ExtendedPdo.php(523): Aura\Sql\ExtendedPdo->perform('SELECT * FROM `...', Array) #2 /home/u12345p12345/domains/mydomain.com/public_html/user/plugins/usrv/plugin.php(86): Aura\Sql\ExtendedPdo->fetchObjects('SELECT * FROM `...') #3 /home/u12345p12345/domains/mydomain.com/public_html/includes/functions-plugins.php(622): usrv_do_page() #4 /home/u12345p12345/domains/mydomain.com/public_html/admin/plugins.php(8): yourls_plugin_admin_page('usrv') #5 {main} thrown in /home/u12345p12345/domains/mydomain.com/public_html/includes/vendor/aura/sql/src/ExtendedPdo.php on line 748
```
The complete webserver logs of the above events are:
```
[Fri Jul 24 13:35:47.440374 2020] [lsapi:notice] [pid 32666:tid 140199719814912] [client 82.169.x.xxx:63135] [host mydomain.com] Backend log: PHP Warning:  copy(assets/srv.php): failed to open stream: No such file or directory in /home/u12345p12345/domains/mydomain.com/public_html/user/plugins/usrv/plugin.php on line 420\n, referer: https://mydomain.com/admin/plugins.php?success=deactivated\
.............................
[Fri Jul 24 13:39:07.645709 2020] [lsapi:error] [pid 32266:tid 140199501604608] [client 82.169.x.xxx:63451] [host mydomain.com] Backend fatal error: PHP Fatal error:  Uncaught PDOException: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'u12345p12345_yourls.usrv' doesn't exist in /home/u12345p12345/domains/mydomain.com/public_html/includes/vendor/aura/sql/src/ExtendedPdo.php:748\nStack trace:\n#0 /home/u12345p12345/domains/mydomain.com/public_html/includes/vendor/aura/sql/src/ExtendedPdo.php(748): PDOStatement->execute()\n#1 /home/u12345p12345/domains/mydomain.com/public_html/includes/vendor/aura/sql/src/ExtendedPdo.php(523): Aura\\Sql\\ExtendedPdo->perform('SELECT * FROM `...', Array)\n#2 /home/u12345p12345/domains/mydomain.com/public_html/user/plugins/usrv/plugin.php(86): Aura\\Sql\\ExtendedPdo->fetchObjects('SELECT * FROM `...')\n#3 /home/u12345p12345/domains/mydomain.com/public_html/includes/functions-plugins.php(622): usrv_do_page()\n#4 /home/u12345p12345/domains/mydomain.com/public_html/admin/plugins.php(8): yourls_plugin_admin_page('usrv')\n#5 {main}\n  thrown in /home/u12345p12345/domains/mydomain.com/public_html/includes/vendor/aura/sql/src/ExtendedPdo.php on line 748\n, referer: https://mydomain.com/admin/plugins.php?success=activated
.............................
[Fri Jul 24 13:44:46.241168 2020] [lsapi:notice] [pid 17046:tid 140199761778432] [client 82.169.x.xxx:63637] [host mydomain.com] Backend log: PHP Warning:  copy(assets/srv.php): failed to open stream: No such file or directory in /home/u12345p12345/domains/mydomain.com/public_html/user/plugins/usrv/plugin.php on line 420\n, referer: https://mydomain.com/admin/plugins.php
```

_Originally posted by @Bogey in https://github.com/joshp23/YOURLS-Expiry/issues/26#issuecomment-663502573_

Edited: removed superfluous log entries

**Comment by joshp23** [comment]
These logs offer no indication as to why the table creation failed. They only indicate that the table cannot be found when looking for it afterwards. I am not encountering that error in my environment.

They do, however, indicate another error altogether, that `assets/srv.php` cannot be found. 

I do not encounter that error in my environment either, however, I can easily fix it for you.

**Comment by joshp23** [comment]
@Bogey, when you deleted the usrv database tables, did you do it manually, and if so did you _also_ delete the `usrv_init` option? If that option exists, the usrv table _will not_ be created. Therefore, it is good to set the option for the usrv table to be removed automatically when the plugin is disabled if you are going to delete it at all.

**Comment by joshp23** [comment]
The error reporting `srv.php` not being found should be fixed with a7b1af75a4face13ac70245e81f6ffaba5df5a33 in 2.3.3 onward.  
The same commit removed the `usrv_init` option, which was redundant anyway. This should fix the database not being created problem encountered after deleting the table manually.

Also, the log above contains the following:
```
SQL: SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES
    		WHERE TABLE_NAME = 'usrv'
    		AND ENGINE = 'MyISAM' LIMIT 1 (0.00636 s)
```
This indicates that an outdated version of this plugin (2.3.0) was being used when the error was reported, as this code was removed with 39f84b410a41635015624adc749f2ce9ce8dcaba. Please make sure that you are using the latest version of the code before reporting an issue. 

Note: When there is an update to the third number, or `x` in `0.0.x`, this indicates a bug fix. This is a feature of [semantic versioning.](https://semver.org/)
